### ISSUE TITLE
Prior fix to outline-regexp prevents code folding

### ISSUE THREAD

**Issue by jeffvalk** [clojure-emacs/clojure-mode#684]
`outline-regexp` was changed in b1ea6de to address #550. The updated regexp deleted one pattern too many.

**tl;dr:** In order to both fix the issue described in #550 _and_ enable unified comment and code folding using [bicycle][b] (or a similar package), `outline-regexp` should be changed from `;;;;* ` to `;;;;* \\|(`.

`bicycle` uses `outline-regexp` to identify code blocks as well as headings. It's just a light wrapper that unifies two built-in folding packages:`outline-minor-mode` (for comments) and `hideshow` (for code). Removing the alternate pattern `\\|(` in `outline-regexp` prevents `bicycle` from finding top-level code blocks to fold; it only permits comment folding. The alternate pattern to match top-level code blocks can be added back to `outline-regexp` without effecting the fix for `outline-insert-heading`:

```elisp
;; Preserve the fix to issue #550, while enabling code + comment folding
(setq-local outline-regexp ";;;;* \\|(") 
```

[b]: https://github.com/tarsius/bicycle/

_Originally posted by @jeffvalk in https://github.com/clojure-emacs/clojure-mode/issues/550#issuecomment-2526671817_

**Comment by bbatsov** [comment]
@jeffvalk Long time, no see!

I'm fine your proposal - feel free to open a PR with the suggested change.

**Comment by jeffvalk** [comment]
Hey @bbatsov -- hope all is well!

PR coming up...

**Comment by jeffvalk** [comment]
It's worth mentioning that this change does alter the behavior of `outline-promote` and `outline-demote`...though not meaningfully. These functions didn't work correctly in `clojure-mode` before the change, and they don't after either. :smile: Of course, they also don't work correctly in `emacs-lisp-mode`, so presumably they're just not something anyone uses in a Lisp.

**Comment by bbatsov** [comment]
Thanks for all the details and the quick fix! 🙇
